Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/common/taimaba/TaimabaActions.java

```diff
@@ -0,0 +1,122 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.github.adamantcheese.chan.core.site.common.taimaba;
+
+import com.github.adamantcheese.chan.core.site.SiteAuthentication;
+import com.github.adamantcheese.chan.core.site.common.CommonSite;
+import com.github.adamantcheese.chan.core.site.common.MultipartHttpCall;
+import com.github.adamantcheese.chan.core.site.http.DeleteRequest;
+import com.github.adamantcheese.chan.core.site.http.DeleteResponse;
+import com.github.adamantcheese.chan.core.site.http.Reply;
+import com.github.adamantcheese.chan.core.site.http.ReplyResponse;
+import org.jsoup.Jsoup;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.HttpUrl;
+import okhttp3.Response;
+
+import static android.text.TextUtils.isEmpty;
+
+public class TaimabaActions extends CommonSite.CommonActions {
+    public TaimabaActions(CommonSite commonSite) {
+        super(commonSite);
+    }
+
+    @Override
+    public void setupPost(Reply reply, MultipartHttpCall call) {
+        call.parameter("board", reply.loadable.board.code);
+
+        if (reply.loadable.isThreadMode()) {
+            call.parameter("thread", String.valueOf(reply.loadable.no));
+        }
+
+        //call.parameter("task", "post");
+        call.parameter("password", reply.password);
+        call.parameter("name", reply.name);
+        call.parameter("email", reply.options);
+
+        if (!isEmpty(reply.subject)) {
+            call.parameter("subject", reply.subject);
+        }
+
+        call.parameter("body", reply.comment);
+
+        if (reply.file != null) {
+            call.fileParameter("file", reply.fileName, reply.file);
+        }
+
+        if (reply.spoilerImage) {
+            call.parameter("spoiler", "on");
+        }
+    }
+
+    @Override
+    public boolean requirePrepare() {
+        return false;
+    }
+
+    /*@Override
+    public void prepare(MultipartHttpCall call, Reply reply, ReplyResponse replyResponse) {
+        TaimabaAntispam antispam = new TaimabaAntispam(
+                HttpUrl.parse(site.resolvable().desktopUrl(reply.loadable, null)));
+        antispam.addDefaultIgnoreFields();
+        for (Map.Entry<String, String> e : antispam.get().entrySet()) {
+            call.parameter(e.getKey(), e.getValue());
+        }
+    }*/
+
+    @Override
+    public void handlePost(ReplyResponse replyResponse, Response response, String result) {
+        Matcher auth = Pattern.compile("\"captcha\": ?true").matcher(result);
```

Comment:
Patterns can be made final static so we don't create the every time anew.

Reply:
i think this section could be simplified as 420chan doesn't use a captcha system, but some kind of anti spam.
https://420chan.org/blog/2013/12/18/captcha-free-anti-spam-tech-now-live/

Reply:
If it won't be used then you can just remove it.